src/OrleansProviders/SQLServer/CreateTables.sql

```diff
@@ -1,113 +1,1440 @@
+		INSERT INTO [OrleansMembershipVersionTable]
+		(
+			[DeploymentId],
+			[Timestamp],
+			[Version]	    
+		)
+		SELECT	
+			@deploymentId,
+			SYSUTCDATETIME(),
+			@version        
+		WHERE NOT EXISTS
+		(
+			SELECT 1
+			FROM OrleansMembershipVersionTable
+			WHERE [DeploymentId] = @deploymentId AND @deploymentId IS NOT NULL
+		);
+                                        
+		IF @@ROWCOUNT > 0
+		BEGIN
+			COMMIT TRANSACTION;
+			SELECT CAST(1 AS BIT);
+		END
+		ELSE
+		BEGIN
+			ROLLBACK TRANSACTION;
+			SELECT CAST(0 AS BIT);
+		END',
+		N''
+	);
+
+	INSERT INTO [OrleansQuery]([Key], [Query], [Description])
+	VALUES
+	(
+		'InsertMembershipKey',
+		N'SET NOCOUNT ON;
+		BEGIN TRANSACTION; --  @@TRANCOUNT = 0 -> +1.
+		-- There is no need to check the condition for inserting
+		-- as the necessary condition with regard to table membership
+		-- protocol is enforced as part of the primary key definition.
+		-- Inserting will fail if there is already a membership
+		-- row with the same
+		-- * [DeploymentId] = @deploymentId
+		-- * [Address]		= @address
+		-- * [Port]			= @port
+		-- * [Generation]	= @generation
+		--
+		-- For more information on table membership protocol insert see at
+		-- http://dotnet.github.io/orleans/Runtime-Implementation-Details/Runtime-Tables.html and at
+		-- https://github.com/dotnet/orleans/blob/master/src/Orleans/SystemTargetInterfaces/IMembershipTable.cs
+		INSERT INTO [OrleansMembershipTable]
+		(
+			[DeploymentId],
+			[Address],
+			[Port],
+			[Generation],
+			[HostName],
+			[Status],
+			[ProxyPort],
+			[Primary],
+			[RoleName],
+			[InstanceName],
+			[UpdateZone],
+			[FaultZone],
+			[SuspectingSilos],
+			[SuspectingTimes],
+			[StartTime],
+			[IAmAliveTime]
+		)
+		VALUES
+		(
+			@deploymentId,
+			@address,
+			@port,
+			@generation,
+			@hostName,
+			@status,
+			@proxyPort,
+			@primary,
+			@roleName,
+			@instanceName,
+			@updateZone,
+			@faultZone,
+			@suspectingSilos,
+			@suspectingTimes,
+			@startTime,
+			@iAmAliveTime
+		);
+
+		IF @@ROWCOUNT = 0 ROLLBACK TRANSACTION;
+
+		IF @@TRANCOUNT > 0
+		BEGIN
+			-- The transaction has not been rolled back. The following
+			-- update must succeed or else the whole transaction needs
+			-- to be rolled back.
+			UPDATE [OrleansMembershipVersionTable]
+			SET
+				[Timestamp]	= SYSUTCDATETIME(),
+				[Version]	= @version
+			WHERE
+				([DeploymentId]	= @deploymentId AND @deploymentId IS NOT NULL)
+				AND ([ETag]		= @versionEtag AND @versionEtag IS NOT NULL);
+
+			-- Here the rowcount should always be either zero (no update)
+			-- or one (exactly one entry updated). Having more means that multiple
+			-- lines matched the condition. This should not be possible, but checking
+			-- only for zero allows the system to function and there is no harm done
+			-- besides potentially superfluous updates.
+			IF @@ROWCOUNT = 0 ROLLBACK TRANSACTION;
+		END
+
+		IF @@TRANCOUNT > 0
+		BEGIN
+			COMMIT TRANSACTION;
+			SELECT CAST(1 AS BIT);
```

Comment:
As stated eariler, I chose to be specific about this. One might get the same information also via "rows affected" or other means, but it would be more fragile as any other measure would rely on observing the success indirectly.

Notable here is that rather than returning just ``TRUE`` or ``FALSE`` anything else could be provided too. For instance, a reason for failure, like ``etag`` mismatch-